cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/file/FileSetDataset.java

```diff
@@ -139,7 +139,8 @@ static Location determineBaseLocation(DatasetContext datasetContext, CConfigurat
 
   private Location determineOutputLocation() {
     String outputPath = FileSetArguments.getOutputPath(runtimeArguments);
-    return outputPath == null ? null : createLocation(outputPath);
+    // TODO: think about all cases of this change (for instance, FileSet as output of MR)
```

Comment:
yeah this seems risky. Would it cause every map/reduce that does not configure an output path and does not use dynamic partitioner to write to the base location?